code/hud/hudsquadmsg.cpp

```diff
@@ -2078,6 +2114,60 @@ void hud_squadmsg_ship_command()
 	}
 }
 
+void hud_squadmsg_msg_general()
+{
+	int k;
+
+	Num_menu_items = 0;
+	for (size_t order_id = 0; order_id < Player_orders.size(); order_id++) {
+		Assert(Num_menu_items < MAX_MENU_ITEMS);
+
+		if (Player_orders[order_id].lua_id <= 0) {
+			continue;
+		}
+
+		auto lua_porder = ai_lua_find_player_order(Player_orders[order_id].lua_id);
+
+		//If it's not a general order then do not add it.
+		if (!lua_porder->generalOrder) {
+			continue;
+		}
+
+		//If it's not part of the selected category then do not add it.
+		if (lua_porder->category != Lua_sqd_msg_cat) {
+			continue;
+		}
```

Comment:
This is player_order number of string comparisons each frame.
Not pretty, but it _might_ be sufficient for now.

Reply:
Hmm. Definitely worth some thought and maybe a follow-up.